server/bin/pbench-server-erase.sh

```diff
@@ -0,0 +1,69 @@
+#! /bin/bash
+
+# Erase Pbench Server 1.0 state completely. This is a simple and "brute force"
+# sequence with few "blade guards", so be careful! (You have been warned!)
+#
+# 1. Delete the Elasticsearch indices and templates using the configured
+#    prefix value;
+# 2. Delete the configured PostgreSQL database;
+# 3. Delete the ARCHIVE, INCOMING, and RESULTS file system trees
+#
+# NOTE: The Pbench server should be *stopped* previously as root, and this
+# script should be run under `su --login pbench`.
+
+es_host=$(pbench-config host elasticsearch)
+es_port=$(pbench-config port elasticsearch)
+
+if [[ -z ${es_host} || -z ${es_port} ]] ;then
+    echo "Host is missing Elasticsearch connection data" >&2
+    exit 1
+fi
+
+elastic="http://${es_host}:${es_port}"
+
+es_prefix=$(pbench-config index_prefix Indexing)
+
+if [[ -z ${es_prefix} ]] ;then
+    echo "Host is missing Elasticsearch prefix" >&2
+    exit 1
+fi
+
+# Clear out the associated Elasticsearch indices and templates
+
+echo "Clearing Elasticsearch ${elastic}, prefix '${es_prefix}'"
+
+curl -X DELETE "${elastic}/${es_prefix}.*"
+curl -X DELETE "${elastic}/_template/${es_prefix}.*"
+
+postgres=$(pbench-config db_uri Postgres)  # postgresql://postgres:secret@localhost/dbname
+ps_server=${postgres%/*}  # e.g., postgresql://postgres:secret@localhost
+ps_db=${postgres##*/}     # e.g., dbname
+
+if [[ -z ${ps_server} || -z ${ps_db} ]] ;then
+    echo "Host is missing Postgres configuration" >&2
+    exit 1
+fi
+
+# Remove the associated SQL database
+
+echo "Clearing PostgreSQL ${ps_server}, database '${ps_db}'"
+
+psql ${ps_server} -c "drop database ${ps_db}"
+
+archive=$(pbench-config pbench-archive-dir pbench-server)
+incoming=$(pbench-config pbench-incoming-dir pbench-server)
+results=$(dirname ${incoming})/results
+
+if [[ ! -d ${archive} || ! -d ${incoming} || ! -d ${results} ]] ;then
+    echo "Host archive (${archive}), incoming (${incoming}) or results (${results}) is missing" >&2
+    exit 1
+fi
```

Comment:
Are you sure you want to exit after issuing this diagnostic?  (Are you sure that you want to make this check at all?...the `rm -rf` command doesn't care....)

Reply:
I was trying to be excessively paranoid in order to protect against either bad values in the config file, or a messed up environment that breaks `pbench-config` (which is actually pretty easy to do with bad `PYTHONPATH`). If I get values that don't look like directories, I don't want to screw things up.

Reply:
I think you almost succeeded...but, for top marks, you need to do the check and exit before touching Elasticsearch and PostgreSQL....

I don't think you need to be paranoid here:  if these directories don't exist, then `rm` will skip over them...yeah, the check enables you to detect a bad or partially-damaged configuration, but, given that we're trying to clean things out here, I don't think aborting in that case is a feature.  (If the configuration is actually bad, then the `rm` is unlikely to find anything to delete.)